packages/material-ui-styles/src/makeStyles/makeStyles.js

```diff
@@ -189,7 +189,7 @@ function useSynchronousEffect(func, values) {
 
 function makeStyles(stylesOrCreator, options = {}) {
   const {
-    // An explicit value provided by the developers.
+    // alias for classNamePrefix, if provided will listen to theme (required for theme.overrides)
```

Comment:
I'm not sure about "alias", from my perspective, aliases have the same behavior. The `classNamePrefix` is only used in development to help find the source of the generates styles. Ideally, we would have source map support instead.
The `name` has two extra features: 1. theme.overrides 2. global class name when prefixed with Mui.

Reply:
We could get rid of the `name` and use `classNamePrefix` only. It would be a breaking change. This is what I meant with alias. It's not 100% correct but close enough to understand what this is used for.

Reply:
What would feed the class name prefix? How do you get helpful debug class names in production while keeping the same behavior for the theme.overrides and global class names in dev & prod. I believe it's two cases that we already support correctly. I can't find where display name leaks to name.

Reply:
It's really not important to discuss this now. It maps close enough for our internal usage. Implementation details can be discussed if need be.

Reply:
I agree that it shouldn't be handled in this pull request.